doc/sphinx/how-to/how-to-enable-graphics-core22-on-a-device.md

```diff
@@ -0,0 +1,632 @@
+SELF="$( cd -- "$(dirname "$0")/.." ; pwd -P )/usr"
+
+
+# mesa-core22 supports multiarch (eg: i386 and amd64 on an amd64 system, or armhf and arm64 on an arm64 system).
+# To do this mesa-core22 needs to supply multiple versions of the libraries, in different paths
+# These paths are architecture-specific, so are filled in at snap build time
+# @ARCH_TRIPLETS@ will be replaced with “i386-linux-gnu x86_64-linux-gnu” when built on amd64
+# and “armhf-linux-gnu arm64-linux-gnu” when built on arm64
+ARCH_TRIPLETS=( @ARCH_TRIPLETS@ )
+
+
+# VDPAU_DRIVER_PATH only supports a single path, rely on LD_LIBRARY_PATH instead
+for arch in ${ARCH_TRIPLETS[@]}; do
+  LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+$LD_LIBRARY_PATH:}${SELF}/lib/${arch}:${SELF}/lib/${arch}/vdpau
+  LIBGL_DRIVERS_PATH=${LIBGL_DRIVERS_PATH:+$LIBGL_DRIVERS_PATH:}${SELF}/lib/${arch}/dri/
+  LIBVA_DRIVERS_PATH=${LIBVA_DRIVERS_PATH:+$LIBVA_DRIVERS_PATH:}${SELF}/lib/${arch}/dri/
+done
+
+__EGL_VENDOR_LIBRARY_DIRS=${__EGL_VENDOR_LIBRARY_DIRS:+$__EGL_VENDOR_LIBRARY_DIRS:}${SELF}/share/glvnd/egl_vendor.d
+__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS=${__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS:+$__EGL_EXTERNAL_PLATFORM_CONFIG_DIRS:}${SELF}/share/egl/egl_external_platform.d
+VK_LAYER_PATH=${VK_LAYER_PATH:+$VK_LAYER_PATH:}${SELF}/share/vulkan/implicit_layer.d/:${SELF}/share/vulkan/explicit_layer.d/
+XDG_DATA_DIRS=${XDG_DATA_DIRS:+$XDG_DATA_DIRS:}${SELF}/share
+
+
+# These are in the default LD_LIBRARY_PATH, but in case the snap dropped it inadvertently
+if [ -d "/var/lib/snapd/lib/gl" ] && [[ ! ${LD_LIBRARY_PATH} =~ (^|:)/var/lib/snapd/lib/gl(:|$) ]]; then
+  LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/var/lib/snapd/lib/gl
+fi
+
+if [ -d "/var/lib/snapd/lib/glvnd/egl_vendor.d" ]; then
+  # This needs to be prepended, as glvnd goes depth-first on these
+  __EGL_VENDOR_LIBRARY_DIRS=/var/lib/snapd/lib/glvnd/egl_vendor.d:${__EGL_VENDOR_LIBRARY_DIRS}
+fi
+
+if [ -d "/var/lib/snapd/lib/vulkan/icd.d" ]; then
+  XDG_DATA_DIRS=${XDG_DATA_DIRS}:/var/lib/snapd/lib
+fi
+
+if [ -d "/var/lib/snapd/lib/gl/vdpau" ]; then
+  LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/var/lib/snapd/lib/gl/vdpau
+fi
+
+export LD_LIBRARY_PATH
+# Mesa-specific environment variable pointing to directory containing $VENDOR_dri.so files
+export LIBGL_DRIVERS_PATH
+export LIBVA_DRIVERS_PATH
+# glvnd-specific environment variable pointing to directory containing ICD descriptor files
+export __EGL_VENDOR_LIBRARY_DIRS
+# EGL external platform interface specific environment
+export __EGL_EXTERNAL_PLATFORM_CONFIG_DIRS
+export VK_LAYER_PATH
+export XDG_DATA_DIRS
+
+exec "$@"
+```
+
+This script has exported the proper paths via environment variables so that
+any snap that connects to our provider snap can properly resolve these
+libraries.
+
+### i386 Support
+If you also need to provide your snap for `i386`, you will need to provide
+a few extra things in the `graphics-core22-provider-wrapper`.
+An example from `mesa-core22` adds these lines:
+
```

Comment:
It's highly likely that the consumers of this guide are going to be ARM-based vendors, so i386 support isn't likely to be very interesting to them. Maybe this instead?
```suggestion
### Multiarch Support
If the vendor drivers provide libraries for multiple sub-architectures (for example, 32-bit and 64-bit ARM) your provider snap can supply both. In this case, you will need to include all provided arch triples in the `ARCH_TRIPLETS` array in the above script (for example: `ARCH_TRIPLETS=( armhf-linux-gnu arm64-linux-gnu )`), and will need to set a few more environment variables in the `graphics-core22-provider-wrapper`.
An example from `mesa-core22` (which provides both `x86_64-linux-gnu` and `i386-linux-gnu`) adds these lines:

```